src/pages/2023-12/next-js-app-router-cache/index.mdx

```diff
@@ -0,0 +1,388 @@
+At build time, the RSC Payload and HTML will be cached in the Full Route Cache. This means that when the user runs the application at run-time to access `/page`, they will be served from the Full Route Cache, resulting in faster page loads.
+
+##### Opting out
+
+If our content requires frequent changing or the feed needs to be personalized, we can choose to opt out of the full route cache.
+
+Simply change the options parameter from `{cache: "force-cache"}` to `{cache: "no-cache"}`.
+
+The full route cache is dependent upon the data cache. Opting out of data cache opts us out of the full route cache as well.
+
+The diagram below demonstrates the step-by-step process of how full route cache works.
+
+![full-route-cache](/articleAssets/2023-12/next-js-app-router-cache/full-route-cache.png)
+
+#### Router Cache
+
+Router cache resides on the client side. It's responsible for storing the RSPC for the duration of the user session or until the page is refreshed.
+It caches the pages the user navigates through during their session, making it faster to load content when they navigate back to previously visited routes.
+
+Next.js can also prefetch content for other linked routes that the user might navigate to next. This prefetching behavior is part of the Router Cache's functionality, further improving navigation speed.
+
+The router cache is a trade-off between performance and real-time data freshness.
+
+The data is stored in the browser's cache. It is not possible to opt-out of router cache.
+
+```jsx {10,11,12}
+export default async function Page() {
+  const blogData = await getBlogList();
+
+  return (
+    <div>
+      <h1>Blog Posts</h1>
+      <ul>
+        {blogData.map((post) => (
+          <li key={post.slug}>
+            <Link href={`/blog/${post.slug}`}>
+              <a>{post.title}</a>
+            </Link>
+            <p>{post.excerpt}</p>
+          </li>
+        ))}
+      </ul>
+    </div>
+  );
+}
+```
+
+If we visit the route `/blog` by clicking on the link and navigate back to `/page` without refreshing, it will return the same blog posts, even if a new blog post was added during that time.
+This is simply because by default Next.js is caching the visited routes.
+
+```jsx {10}
+export default async function Page() {
+  const blogData = await getBlogList();
+
+  return (
+    <div>
+      <h1>Blog Posts</h1>
+      <ul>
+        {blogData.map((post) => (
+          <li key={post.slug}>
+            <Link href={`/blog/${post.slug}`} prefetch={false}>
+              <a>{post.title}</a>
+            </Link>
+            <p>{post.excerpt}</p>
+          </li>
+        ))}
+      </ul>
+    </div>
+  );
+}
+```
+
+
+The diagram below demonstrates the step-by-step process of how the router cache works.
+
+![router-cache](/articleAssets/2023-12/next-js-app-router-cache/router-cache.png)
+
+#### Cache interactions
```

Comment:
I don't think this section is needed and instead the important information can be moved up into each individual route section.